### PR TITLE
Post export bug fixing

### PR DESCRIPTION
**Pull request by remixz**
* Refactor CSV export to use [`csv-write-stream`](https://github.com/maxogden/csv-write-stream) (Fixes GH-104)
* Use `Accept` header when no extension is provided (defaults to .json)

### CODE REVIEW

**Comment by isaacs** [comment]
Still a problem.

Create a post with this body: `double " quote` and then export, and parse with any CSV parser.

The best approach here is to find and use a CSV encoding module, not do it inline by hand.

**Comment by remixz** [comment]
Hm, that works fine for me... the `"` is escaped by `twitter-text`, so it just shows as `&quot;`. In any case though, I've refactored it with [`csv-write-stream`](https://github.com/maxogden/csv-write-stream), since it's definitely better to have a csv parser with tests.

**Comment by isaacs** [comment]
Aha!  That's a good point, html encoding does save us there.

Still, yes, a proper CSV parser probably has lots of other edge cases you and I haven't thought of :)

File: lib/profile.js [hunk 1]
```diff
@@ -300,35 +305,21 @@ exports.exportPosts = function (request, reply) {
...
+      writer.pipe(concat(function (csv) {
+        reply(csv).type('text/csv');
+      }));
+
       posts.forEach(function (post) {
-        var val = [];
-        Object.keys(post).forEach(function (key) {
-          var i = keys.indexOf(key);
-          val[i] = post[key];
+        var vals = Object.keys(post).map(function (key) {
```
Comment (isaacs) [inline lib/profile.js]: This should use the same list of keys that you set in the headers.

Otherwise, when/if we add new keys, it'll break in funky ways.

Comment (isaacs) [inline lib/profile.js]: Ie, instead of doing Object.keys() here, and a literal list above, set `var keys = [ ... ]` and use that in both places.

File: lib/profile.js [hunk 2]
```diff
@@ -284,6 +285,10 @@ exports.addPhone = function (request, reply) {
 };
 
 exports.exportPosts = function (request, reply) {
+  if (request.params.ext !== 'json' && request.params.ext !== 'csv') {
```
Comment (isaacs) [inline lib/profile.js]: I'm not particularly religious about RESTful semantics or anything, but could it use the `accept` header if no ext is provided?

File: lib/profile.js (hunk 1 above)
Comment (remixz) [inline lib/profile.js]: :+1: Done!

File: lib/profile.js (hunk 2 above)
Comment (remixz) [inline lib/profile.js]: Done as well. 345dc7c

**Comment by isaacs** [comment]
Landed, thanks!
